tests.ts

```diff
@@ -0,0 +1,67 @@
+const functsFromMain = require('./main');
+const assert = require('assert');
+
+// test for formReadRes function
+describe("formReadRes", () => {
+    // execute function
+    functsFromMain.formReadRes('data.txt')('result.txt')("l")
+    it("given 'data.txt' contains the same number of symbol 'l' as written in the final file", () => {
+        // read original file 
+        functsFromMain.readFile('data.txt', 'utf8', (err: string, data: string) => {
+            if (err) {
+                throw err
+            } else {
+                // calculate number of symbol 'l' in text of the original file 
+                const resNumOfSymbs = functsFromMain.numOfSymbols("l")(data)
+                // read file with result of calculations 
+                functsFromMain.readFile('result.txt', 'utf8', (err: string, data: string) => {
+                    if (err) {
+                        throw err
+                    } else {
+                        // number of symbols to integer 
+                        const numOfSymbEInFile = parseInt(data)
+                        // check if number of 'l' symbol 
+                        // is the same as written in the result file
+                        assert.strictEqual(resNumOfSymbs, numOfSymbEInFile)
```

Comment:
This whole test does not check anything. On the line 22, change `numOfSymbEInFile` to ANY other value, and the test still succeeds, meaning it doesn't work. Note: in mocha, tests succeed, unless they encounter assertions during their immediate execution OR we make them deferred tests with `done` argument. None of these things are done here.

You can also notice an error AFTER the test suite is completed, about not finding "result.txt". This should immediately tell you that something is wrong with order of execution and test completion.